automation/prepare-release.py

```diff
@@ -68,24 +57,16 @@ def fatal_err(msg):
 
 step_msg(f"Preparing release {next_version_full}")
 
-# 2. Switch to proper branch and do some consistency checks.
+# 2. Create a new branch based on the branch we want to release from.
 
-if run_cmd(["git", "status", "--porcelain"], capture_output=True).stdout:
-    fatal_err("This branch has un-commited or staged files.")
+ensure_working_tree_clean()
 
 step_msg(f"Updating remote {remote}")
-run_cmd(["git", "remote", "update", remote])
-
-step_msg(f"Checking remote {remote} agrees with us")
-remote_hash = run_cmd(["git", "rev-parse", f"{remote}/{base_branch}"], capture_output=True, text=True).stdout
```

Comment:
I removed the consistency check here, as we are creating a branch from the remote repo anyway (which is up-to-date since we `git remote update` earlier).